pkg/builder3/openapi.go

```diff
@@ -436,3 +438,30 @@ func (o *openAPI) toSchema(name string) (_ *spec.Schema, err error) {
 		}, nil
 	}
 }
+
+// wrapRefs wraps OpenAPI V3 Schema refs that contain sibling elements.
+// AllOf is used to wrap the Ref to prevent references from having sibling elements
+// Please see https://github.com/kubernetes/kubernetes/issues/106387#issuecomment-967640388
+func wrapRefs(schema *spec.Schema) *spec.Schema {
+	walker := schemamutation.Walker{
+		SchemaCallback: func(schema *spec.Schema) *spec.Schema {
+			orig := schema
+			clone := func() {
+				if orig == schema {
+					schema = new(spec.Schema)
+					*schema = *orig
+				}
+			}
+			if schema.Ref.String() != "" && (schema.Nullable != false || schema.Default != nil || len(schema.Extensions) > 0) {
```

Comment:
I'm questioning if we have a better way to do these checks.

Reply:
you could check if, when removing the ref, you have an empty schema. Or you could copy the ref to an empty schema and compare them? I'm assuming that what you're actually checking is that there is no other field along with the ref, your condition will need to be updated anytime we add such a new thing in our openapi.